compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -507,6 +507,63 @@ SortOp::reifyResultShapes(OpBuilder &b,
       .reifyResultShapes(b, reifiedReturnShapes);
 }
 
+LogicalResult SortOp::canonicalize(SortOp sortOp, PatternRewriter &rewriter) {
+  // To avoid problems in dispatches associated with unused results, prune them
+  // here.
+  auto loc = sortOp->getLoc();
+  auto operands = sortOp.getOutputs();
+  auto results = sortOp.getResults();
+
+  // If there are no results, bufferization has happened.
+  // Do not try to prune results anymore.
+  if (results.size() == 0) {
+    return failure();
+  }
+
+  Block &block = sortOp.getRegion().front();
+  auto blockArgs = block.getArguments();
+  auto used = [&](Value result) { return !result.use_empty(); };
+  SmallVector<bool> resUsed = llvm::map_to_vector(results, used);
+  SmallVector<bool> argsUsed = llvm::map_to_vector(blockArgs, used);
```

Comment:
```suggestion
 SmallVector<bool> isResultIdxUsed = llvm::map_to_vector(results, used);
SmallVector<bool> isArgIdxUsed = llvm::map_to_vector(blockArgs, used);
``` Looks more descriptive.